src/webui/www/private/views/rss.html

```diff
@@ -503,9 +503,12 @@
                     if ((rssFeedRows.length - 1) === flattenedResp.length) {
                         match = true;
                         for (let i = 0; i < flattenedResp.length; ++i) {
-                            if (((flattenedResp[i].uid ? flattenedResp[i].uid : "") !== rssFeedRows[i + 1].full_data.dataUid)
-                                || (flattenedResp[i].fullName !== rssFeedRows[i + 1].full_data.dataPath)
-                                || (flattenedResp[i].url !== rssFeedRows[i + 1].full_data.dataUrl)) {
+                            const newData = flattenedResp[i];
+                            const existingData = rssFeedRows[i + 1].full_data;
+                            if ((newData.fullName !== existingData.dataPath)
+                                // folders don't have the following properties
+                                || (!newData.isFolder && (newData.uid !== existingData.dataUid))
+                                || (!newData.isFolder && (newData.url !== existingData.dataUrl))) {
```

Comment:
```suggestion
                                || (!newData.isFolder && (newData.uid !== existingData.dataUid)
                                    || (newData.url !== existingData.dataUrl)) {
```

Reply:
We need a folder check for the `dataUrl` as well.

Reply:
Ah I see now that you're combining the two checks. I'm not sure that's necessarily easier to read than the current version.

Reply:
>I'm not sure that's necessarily easier to read than the current version.

As long as it can pass the CI checks, I don't mind you rearrange the code a bit or use another temp variable.
Or maybe just put them on the same line:
```js
(!newData.isFolder && ((newData.uid !== existingData.dataUid) || (newData.url !== existingData.dataUrl))) {
```
The main point is to reduce redundant predicates/checks.

Reply:
I've adjusted these checks so that the intention is clearer.